src/ray/raylet/local_task_manager.cc

```diff
@@ -71,6 +71,21 @@ LocalTaskManager::LocalTaskManager(
       sched_cls_cap_interval_ms_(sched_cls_cap_interval_ms),
       sched_cls_cap_max_ms_(RayConfig::instance().worker_cap_max_backoff_delay_ms()) {}
 
+void LocalTaskManager::UpdateCpuRequests(const std::shared_ptr<internal::Work> &work,
```

Comment:
This method must be called when a work is added to a dispatch queue, or when a work is removed from it. This is error prone. I am thinking, what about making a class to wrap the work like this?

```cc

class LocalTaskManager {

  class CpuAddingWork {
    explicit CpuAddingWork(LocalTaskManager &local_task_manager,
                           std::shared_ptr<internal::Work> work)
        : local_task_manager_(local_task_manager), work_(work) {
      local_task_manager_.num_total_cpus_ += GetCpuRequests();
    }

    ~CpuAddingWork() { local_task_manager_.num_total_cpus_ -= GetCpuRequests(); }

    internal::Work *get() const { return work_.get(); }
    internal::Work *operator->() const { return work_.get(); }
    internal::Work &operator*() { return *work_; }

   private:
    double GetCpuRequests() {
      // FixedPoint to Double
      return work_->task.GetTaskSpecification()
          .GetRequiredResources()
          .Get(scheduling::ResourceID::CPU())
          .Double();
    }
    LocalTaskManager &local_task_manager_;
    std::shared_ptr<internal::Work> work_;
  };

  int num_total_cpus_ = 0;
  absl::flat_hash_map<SchedulingClass, std::deque<CpuAddingWork>> tasks_to_dispatch_;
};
```